src/Disks/IO/AsynchronousReadIndirectBufferFromRemoteFS.cpp

```diff
@@ -152,12 +154,16 @@ bool AsynchronousReadIndirectBufferFromRemoteFS::nextImpl()
         CurrentMetrics::Increment metric_increment{CurrentMetrics::AsynchronousReadWait};
         Stopwatch watch;
         {
-            size = prefetch_future.get();
+            size_t offset;
+            std::tie(size, offset) = prefetch_future.get();
+            assert(offset < working_buffer.size());
             if (size)
             {
                 memory.swap(prefetch_buffer);
                 set(memory.data(), memory.size());
                 working_buffer.resize(size);
```

Comment:
maybe we need `size - offset`?

Reply:
No, becuase first `offset` bytes must be skipped. We need last `working_buffer.size() - offset` bytes, so this is what is done in the next few lines -- we get correct part.

Reply:
Simplified this part.